src/scripts/main.js

```diff
@@ -2,10 +2,23 @@
 
 function waitFor(element, eventName) {
   // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, function handler() {
```

Comment:
The handler function for the event listener doesn't check if the event has occurred. It's better to add a condition to check if the event has occurred before resolving the promise.